Makefile

```diff
@@ -31,7 +31,9 @@ nns_sc = nns
 
 define sc_template
 $(2)$(1)/$(1)_contract.nef: $(2)$(1)/$(1)_contract.go
-	$(NEOGO) contract compile -i $(2)$(1) -c $(if $(2),$(2),$(1)/)config.yml -m $(2)$(1)/config.json -o $(2)$(1)/$(1)_contract.nef
+	$(NEOGO) contract compile -i $(2)$(1) -c $(if $(2),$(2),$(1)/)config.yml -m $(2)$(1)/config.json -o $(2)$(1)/$(1)_contract.nef --guess-eventtypes --bindings $(2)$(1)/bindings_config.yml
```

Comment:
`--guess-eventtypes`?

Reply:
Yes, and I'm strongly for the usage of `--guessed-eventtypes` flag, because it will properly handle all complicated event types without additional configuration. Although events are pretty simple in these contracts.

Reply:
To me this implies not having event in configuration files. But we have them now. And removing them is a bit scary to me.

Reply:
The only complex case is probably
```
      - name: keys
        type: Array
```
But I think it should be handled fine without guesses as well.

Reply:
It will, but without guesses its type will be `[]any`, because it's impossible to restore the type of array elements from the config without additional config modification:
```
// BindEvent represents "Bind" event emitted by the contract.
type BindEvent struct {
	User []byte
	Keys []any
}
```
And with guesses it has the proper and nice type:
```
// BindEvent represents "Bind" event emitted by the contract.
type BindEvent struct {
	User []byte
	Keys keys.PublicKeys
}
```

Reply:
What's gonna happen if I'm to write
```
      - name: keys
        type: Map
```
and compile with `--guess-eventtypes`?

Reply:
Currently the result with this flag will contain real type of the event parameters, and we have https://github.com/nspcc-dev/neo-go/issues/3027 to be resolved:
```
							// TODO: see the TestCompile_GuessEventTypes, "SC parameter type mismatch" section,
							// do we want to compare with actual.RealType? The conversion code is emitted by the
							// compiler for it, so we expect the parameter to be of the proper type.
```
So the binding result will be:
```
// BindEvent represents "Bind" event emitted by the contract.
type BindEvent struct {
	User []byte
	Keys keys.PublicKeys
}
```
The compiler will emit the conversion code to convert `Keys` to map and execution of the contract will fail.

>  the result will contain real type of the event parameters

This happens because we firstly check the predicted emitted parameter types against the configured one. The predicted emitted parameter type is `Map` in this case, because conversion code is emitted for this parameter. Thus, the check agains config is passed. After that if all parameter checks are passed we emit real parameter type to the binding configuration, and the real type in this case is `keys.PublicKeys`. As a result, we've got it in the resulting binding. I'm not sure whether we need it to be fixed right now, because if we're to solve https://github.com/nspcc-dev/neo-go/issues/3027, then the parameter check won't pass for such event and compilation will fail (because the baes type for `keys.PublicKeys` is an array, not `Map`).

Reply:
The perfect solution would be to store the smartcontract's real type of the parameter (not the one that we'll get after conversion), and if it differs from the config, then take the one that is declared in the config and generate RPC binding with it (do not pay attention to `RealType` in this case). It will work even without https://github.com/nspcc-dev/neo-go/issues/3027 fix. Implement?

Reply:
Fixed.